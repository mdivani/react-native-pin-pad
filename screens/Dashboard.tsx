import axios from 'axios';
import React, { useContext, useState } from 'react';
import { ActivityIndicator, Alert, Pressable } from 'react-native';
import EStyleSheet from 'react-native-extended-stylesheet';
import { View, Text } from '../components/Themed';
import { AuthContext } from '../context/authContext';

export default function Dashboard() {
    const [fetching, setFetching] = useState(false);
    const auth = useContext(AuthContext);

    const logout = async () => {
        try {
            setFetching(true);
            const response = await axios.post(
                "http://localhost:3000/logout",
                {},
                {
                    headers: {Authorization: auth.token},
                },
            );
            setFetching(false);

            if (response.status === 200) {
                auth.setToken("");
            }
        } catch (ex) {
            setFetching(false);
            Alert.alert("Failed to logout", "Please try again later");
        }
    }

    return (
        <View style={styles.container}>
            <Text style={styles.title}>Hi, Welcome</Text>
            <Pressable onPress={logout} style={styles.button}>
                {
                    fetching ?
                    <ActivityIndicator /> :
                    <Text style={styles.buttonTitle}>Logout</Text>
                }
            </Pressable>
        </View>
    );
}

const styles = EStyleSheet.create({
    container: {
        flex: 1,
        alignItems: "center",
        padding: "2rem"
    },
    title: {
        fontSize: "2.5rem",
    },
    button: {
        borderWidth: 1,
        borderColor: "white",
        paddingVertical: "1rem",
        paddingHorizontal: "2rem",
        marginVertical: "2rem",
        width: "100%",
        alignItems: "center",
    },
    buttonTitle: {
        fontSize: "2rem",
    }
});
