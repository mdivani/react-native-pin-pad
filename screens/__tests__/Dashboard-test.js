import * as React from 'react';
import renderer from 'react-test-renderer';
import {render, fireEvent, cleanup} from '@testing-library/react-native';

import Dashboard from '../Dashboard';

beforeEach(cleanup);

it(`renders correctly`, () => {
    const tree = renderer.create(<Dashboard />).toJSON();
    expect(tree).toMatchSnapshot();
});