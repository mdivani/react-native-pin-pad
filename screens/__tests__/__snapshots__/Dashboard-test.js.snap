// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        undefined,
      ]
    }
  >
    Hi, Welcome
  </Text>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          undefined,
        ]
      }
    >
      Logout
    </Text>
  </View>
</View>
`;
